sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-generation.service.ts

```diff
@@ -192,14 +200,30 @@ export class ECertGenerationService {
         END`,
         "stopFullTimeBCFunding",
       )
+      .addSelect(
```

Comment:
Is it not the case of, instead of having 2 subqueries, we start to bring all active restrictions and do the verifications on the in-memory list, which in this case will be "all active restrictions that can impact disbursement"?
I am assuming also the possibles `StopFullTimeDisbursement` and `StopPartTimeDisbursement` will be already excluded by the where condition at line 244, since these records will not even be selected.

Reply:
ya, get the whole list and assign the latest restriction to the bc grants and loans when needed?..mm ok...actually, I was trying to avoid the number of queries. But i can give it a try

Reply:
https://github.com/bcgov/SIMS/pull/1886#discussion_r1171972543 but I have added 
![image](https://user-images.githubusercontent.com/77353155/233228851-3ef6b1f4-c146-4b56-9b9c-7893e27fab6b.png)